src/docdb/tree/DocDBTreeItemBase.ts

```diff
@@ -33,7 +37,12 @@ export abstract class DocDBTreeItemBase<T> implements IAzureParentTreeItem {
     }
 
     public getDocumentClient(): DocumentClient {
-        return new DocumentClient(this.documentEndpoint, { masterKey: this.masterKey });
+        const documentBase = lib.DocumentBase;
+        var connectionPolicy = new documentBase.ConnectionPolicy();
+        const userPreferenceSSLVerify: boolean = vscode.workspace.getConfiguration().get("http.proxyStrictSSL");
+        connectionPolicy.DisableSSLVerification = !userPreferenceSSLVerify && !this.SSLVerify;
```

Comment:
I'd like to get some second opinions on this since it has potential security risks to the user. @whiskeyjay @StephenWeatherford the Cosmos DB emulator always uses self-signed certificates so we have to set connectionPolicy.DisableSSLVerification for it to work. How do you think we should determine when to set DisableSSLVerification?

The 'http.proxyStrictSSL' Prashanth is using here is defined by VS Code

Reply:
Swamped, I'll get back to you.

Reply:
If connectionPolicy.DisableSSLVerification is for the emulator connection only, and setting this flag alone can make things work, I'd show a prompt the first time when a user tries to add the connection explaining the connection will ignore the SSL cert, and ask for their permission to proceed. I believe name "localhost" can be hijacked in some circumstances, causing the extension trying to connect to an unexpected server, that plus ignoring SSL cert can be a problem.